Pull Request on Challenge 2

NabunyaLilian opened the pull request:
Hey guys, this is my latest pull request on challenge, please review my work  I'm open to any feedback.

NabunyaLilian committed:
new changes in procfile

NabunyaLilian committed:
new changes made

NabunyaLilian committed:
new changes made

NabunyaLilian committed:
new changes made

NabunyaLilian committed:
new changes made

NabunyaLilian committed:
new changes made

NabunyaLilian committed:
pylint modifications

NabunyaLilian committed:
new changes PEP3 standard

NabunyaLilian committed:
modification in readme

NabunyaLilian committed:
modification in readme

NabunyaLilian committed:
modification in readme

NabunyaLilian committed:
modification in readme

NabunyaLilian committed:
Pipfile added

NabunyaLilian committed:
add runtime

NabunyaLilian committed:
readme updated

NabunyaLilian committed:
readme updated

NabunyaLilian committed:
add changes in readme

NabunyaLilian committed:
Add new change in readme

NabunyaLilian committed:
Save changes

NabunyaLilian committed:
save changes in readme

NabunyaLilian committed:
add badges in readme

NabunyaLilian committed:
add badges in readme

NabunyaLilian committed:
add badges in readme

NabunyaLilian committed:
add badges in readme

NabunyaLilian committed:
save changes in readme

NabunyaLilian committed:
save changes in readme

NabunyaLilian committed:
save changes in readme

NabunyaLilian committed:
save changes in readme

NabunyaLilian committed:
add changes in readme

NabunyaLilian committed:
add changes in readme

NabunyaLilian committed:
save new changes

NabunyaLilian committed:
Change folder structre

NabunyaLilian committed:
add validation file

NabunyaLilian committed:
add validation file

NabunyaLilian committed:
save feedback changes

NabunyaLilian committed:
update readme

calvinpete commented:
Good work Lillian, though a few things can be done to make it better;
- Modify your readme file with a command for cloning your app using Git.
- In my opinion, the postman collection can be made better. **check out:** https://medium.com/@olotintemitope/how-to-generate-your-api-documentation-in-20-minutes-4e0072f08b94
- Add a link to your app hosted on Heroku. It's what is used to access your API endpoints instead of http://127.0.0.1:5000/
- I checked your coverage and you are missing tests on crucial lines of code. i.e API endpoints and the validation file

NabunyaLilian commented:
okay i will look into that @calvinpete

NabunyaLilian committed:
save new changes

NabunyaLilian committed:
save new changes

NabunyaLilian committed:
update readme file

NabunyaLilian committed:
update readme file

NabunyaLilian committed:
update readme file

NabunyaLilian committed:
Increase on Modularity

* create a file and name it check.py to handle authorization of using a resource when id is already existing.
* use check.py in product_view.py to check if a product_id is already existing then return the product.
* use it also in sale_view.py to check if a sale_id is already existing then return sale record.
* This increases on the modularity of the code.

NabunyaLilian committed:
Include in more validation

* validate products_list_view to check for empty fields and empty fields' values
* validate sales_view.py to check for empty fields and empty fields'values
* validate product_view.py to check for id that doesnot exist when a get
request has been made.
* validate sale_view.py to check for id that doesnot exist when a get
request has been made.

NabunyaLilian committed:
Add config file

*Add config file to setup configurations for different environments

NabunyaLilian committed:
save new changes

NabunyaLilian committed:
update config file

*include in testing environment in the config.file
*apply the testing environment in test_api.py

NabunyaLilian committed:
modify config file

NabunyaLilian committed:
remove challenge 2 travis
